File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    answer = []
    
    zeroCnt = 0
    calCnt = 0
    
    while(True):
        if s=='1': break
        
        for c in s:
            if c=='0': zeroCnt += 1 # 0개수 세기
        
        s = s.replace('0', '')
        s = bin(len(s))[2:] #2진수로 변환
        
        calCnt += 1
    
    answer.append(calCnt)
    answer.append(zeroCnt)
        
    return answer

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(string s) {
    vector<int> answer;
    int repeat_cnt=0;
    int zero_cnt = 0;
    
    while (true) {
        if (s.length() == 1) {
            answer.push_back(repeat_cnt);
            answer.push_back(zero_cnt);
            break;
        }
        
        vector<char> str;
        for (int i=0; i<s.length(); i++) {
            if (s[i] == '0') zero_cnt += 1;
            else str.push_back(s[i]);
        }
        
        int new_str_len = str.size();
        string tmp_str = "";
        while (new_str_len > 0) {
            tmp_str = to_string(new_str_len & 1) + tmp_str;
            new_str_len >>= 1;
        }
        
        s = tmp_str;
        repeat_cnt += 1;
    }
    
    return answer;
}

/*
문자열 벡터에 0을 제거한 1로 만들어진 숫자를 넣음
벡터의 길이를 이진수로 변환함
*/